src/components/viewers/XLSViewer.vue

```diff
@@ -1,116 +1,103 @@
 <template>
   <div class="text-viewer">
     <div class="wrapper">
-      <table v-if="parsed" style="width: 100%;">
+      <table v-if="parsed" style="width: 100%">
         <thead>
-        <tr>
-<!--          <th v-for="(header, key) in content.meta.fields"-->
-<!--              v-bind:key="'header-'+key">-->
-<!--            {{ header }}-->
-<!--          </th>-->
-        </tr>
+          <tr>
+            <th v-for="column in uniqueColumns" :key="column">{{ column }}</th>
+          </tr>
         </thead>
         <tbody>
-        <tr v-for="(row, rowKey) in content"
-            v-bind:key="'row-'+rowKey">
-          <td v-for="(column, columnKey) in Object.keys(content[rowKey])"
-              v-bind:key="'row-'+rowKey+'-column-'+columnKey">
-            <div class="table-cell">
-              {{content[rowKey][column]}}
-            </div>
-          </td>
-        </tr>
+          <tr v-for="(row, rowKey) in content" :key="'row-' + rowKey">
+            <td
+              v-for="column in uniqueColumns"
+              :key="'row-' + rowKey + '-column-' + column"
+            >
+              <div class="table-cell">
+                {{ row[column] || "" }}
+              </div>
+            </td>
+          </tr>
         </tbody>
       </table>
-
-
-
-      <!--      <pre>-->
-      <!--        <code-->
-      <!--          v-if="isText"-->
-      <!--          ref="codeblock"-->
-      <!--          class="codeblock"-->
-      <!--          :class="subtype"-->
-      <!--          v-html="escapeHTMLData(fileData)"-->
-      <!--        />-->
-      <!--        <code-->
-      <!--          v-if="!isText"-->
-      <!--          ref="codeblock"-->
-      <!--          class="codeblock"-->
-      <!--          :class="subtype"-->
-      <!--        >{{ fileData }}</code>-->
-      <!--      </pre>-->
     </div>
   </div>
 </template>
 
 <script>
-
-import StaticViewer from '../../mixins/static-viewer'
-import GetFileProperty from '../../mixins/get-file-property'
-import Request from '../../mixins/request'
-import Papa from 'papaparse';
-import { read, utils} from 'xlsx'
+import StaticViewer from "../../mixins/static-viewer";
+import GetFileProperty from "../../mixins/get-file-property";
+import Request from "../../mixins/request";
+import Papa from "papaparse";
+import { read, utils } from "xlsx";
+import { forEach } from "ramda";
 
 export default {
-  name: 'XLSViewer',
+  name: "XLSViewer",
 
-  mixins: [
-    StaticViewer,
-    GetFileProperty,
-    Request
-  ],
+  mixins: [StaticViewer, GetFileProperty, Request],
 
   props: {
     pkg: {
       type: Object,
-      default: () => {}
+      default: () => {},
     },
     idx: {
       type: Number,
-      default: 0
-    }
+      default: 0,
+    },
   },
 
   data() {
     return {
-      fileData: '',
+      fileData: "",
       content: {},
```

Comment:
Since the `content` data type in this component is an array, it could be initialized to an empty array.

Reply:
good catch - ty!